crates/router/src/core/payments.rs

```diff
@@ -189,11 +178,24 @@ where
                 .await?
             }
         };
+
         if should_delete_pm_from_locker(payment_data.payment_intent.status) {
             vault::Vault::delete_locker_payment_method_by_lookup_key(state, &payment_data.token)
                 .await
         }
+    } else {
+        operation
+            .to_update_tracker()?
```

Comment:
In PR description it is mentioned that update trackers would be called before calling the connector . But here it is called after calling the connector .

Reply:
This is a normal flow where connector does not need to be called, this is the else branch ( when we don't have connector). These include flows like `payments_update`,  so we just go with the normal flow.